packages/cli/core/src/lib/snapshot/snapshot.spec.ts

```diff
@@ -2,6 +2,7 @@ import {
   ResourceSnapshotsModel,
```

Comment:
General comment on the change in this spec, relative to the change in its associated source file.

I think the change in these specs is testing the mocks, but not the actual change in the source code.
Indeed, if you read it out, you'll see that you mock a method and make the mock throw an error, and then test that the caller of this method does throw.

So, in essence, here we are asserting that the method is called by testing its mocked behavior.

This could be simplified by only testing what's actually happening in this method from a 'sandboxed' perspective:
Do I call the method, and do I call it with the right parameters?

Because the method is properly unit-tested, this should offer proper coverage and simplify those tests.

Reply:
I see what you are saying, but should the method checks at least if it resolves or rejects based on the resource access precondition (regardless of the error thrown)?

Reply:
I think this is not necessary because we don't have error handling logic in this method: we just bubble up the error up the stack here.